### ISSUE TITLE
Plasma's global menu stopped working (it works in flatpak version 0.11.7 but it doesn't in 1.0.0)

### ISSUE THREAD

**Issue by kupiqu** [flatpak/flatpak#2012]
## Linux distribution and version
KDE neon User Edition 18.04

## Flatpak version
1.0.0 (but not that it works fine in version 0.11.7)

## Description of the problem
The application menu is not exported to plasma

## Steps to reproduce
Start a flatpak application and see that the menu is not exported even if configured as such

**Comment by kupiqu** [comment]
At first I thought it was a KDE issue (https://bugs.kde.org/show_bug.cgi?id=397625) but given that it works for a previous version of flatpak, now I'm not that sure...

**Comment by kupiqu** [comment]
I could test version 0.99.3 and this bug appllies to that version as well

**Comment by kupiqu** [comment]
I could track it down to the transition between 0.11.8.3 (which works) and 0.99.1 (which doesn't)

**Comment by alexlarsson** [comment]
So, i don't actually know how the global menu works. Is that some dbus api?

**Comment by alexlarsson** [comment]
Ah, yes. I think I can guess what caused this: https://github.com/flatpak/flatpak/pull/1799
In particular this one:
  https://github.com/flatpak/flatpak/pull/1799/commits/8c22b4400d0347a0c5647ce0767b7bfef98ee94b

We stopped inheriting permission requests from the runtime. The apps need to list the permissions themselves. This was done because we started listing and pre-enforcing the permissions and it became very hard to follow for users if permissions affect other things than the one you install.

Looking at https://github.com/KDE/flatpak-kde-runtime/blob/master/org.kde.Sdk.json#L45 it seems that the apps need to copy the --filesystem and --talk-name from there (if they are used by the app).

**Comment by alexlarsson** [comment]
And, i'm terribly sorry about this. I didn't realize anything actually depended on this feature.

**Comment by kupiqu** [comment]
I see. I guess this means that every KDE-dependent application needs to add these flags from now on, but how? They don't seem to work when added manually to one of those KDE-dependent applications

**Comment by kupiqu** [comment]
I guess applications need to be recompiled, right? Or, am I missing something?

**Comment by Pointedstick** [comment]
Could we reconsider no longer inheriting permission requests from the runtime? That change makes it hard to have a single consistent set of permissions, and will require every client to re-implement the same stuff, which seems like a lot of work and a violation of DRY.

If this inheritance led to user confusion, perhaps we could work on the presentation and UI rather than removing the feature?

**Comment by kupiqu** [comment]
It doesn't only affect globalmenu but also the window theme (cosmetic if you will), but also, and quite more important, the ui dialogs (e.g., open/save). This is important as now the productivity decays strongly (at least with KDE-dependent applications). Leaving this workload to packagers in a per application basis will surely break the system consistency that flatpak applications reached before (one couldn't distinguish easily between native linux and flatpak packages, but no more)

**Comment by kupiqu** [comment]
> > the ui dialogs (e.g., open/save)

> I need to investigate this further as dialogs work in KDE applications (e.g., kdenlive) but not in KDE-dep applications like Octave

I think this particular thing about the dialogs is an Octave issue (https://github.com/flathub/org.octave.Octave/issues/28)

Globalmenu and application theme consistently fail in KDE apps (e.g., kdenlive) and Octave, though

**Comment by kupiqu** [comment]
> Could we reconsider no longer inheriting permission requests from the runtime? That change makes it hard to have a single consistent set of permissions, and will require every client to re-implement the same stuff, which seems like a lot of work and a violation of DRY.
> 
> If this inheritance led to user confusion, perhaps we could work on the presentation and UI rather than removing the feature?

Dear @alexlarsson, 

Wondering if this will be reconsidered as requested by KDE devs (and users), with a low priority perhaps, or whether it would be better to close the issue as won't fix.

Personally I'm  holding my system with flatpak 0.11.8.3 for this specific reason, and would like to know if it makes sense to continue this way until the issue gets eventually considered, or whether I should rather upgrade flatpak and deal with this for eah application individually. 

Not aiming to make pressure, just want to know if you've already made a decision on this.

Thank you

**Comment by Pointedstick** [comment]
FWIW, we fixed this in KDE by adding the necessary config pieces to all KDE apps (https://cgit.kde.org/flatpak-kde-applications.git/commit/?id=0ccbc716568441922961a8ae372284a9d07be882), which was not too bad because we have all our apps' Flatpak configs in the same repo.
